security/KeePassXC/Portfile

```diff
@@ -42,6 +43,15 @@ depends_lib-append      port:argon2 \
                         port:libsodium \
                         port:zlib
 
+# KeePassXC uses -fstack-protector-strong on Clang [1]. That flag is not
+# available until clang 602 [2] or upstream clang 3.5 [3]
+# [1] https://github.com/keepassxreboot/keepassxc/blob/develop/CMakeLists.txt
+# [2] https://opensource.apple.com/source/clang/clang-600.0.57/src/tools/clang/lib/CodeGen/CodeGenModule.cpp.auto.html
+#     https://opensource.apple.com/source/clang/clang-602.0.53/src/tools/clang/lib/CodeGen/CodeGenModule.cpp.auto.html
+# [3] https://github.com/llvm-project/clang/blob/release_34/lib/CodeGen/CodeGenModule.cpp
+#     https://github.com/llvm-project/clang/blob/release_35/lib/CodeGen/CodeGenModule.cpp
+compiler.blacklist-append {clang < 602} macports-clang-3.4 macports-clang-3.3
+
```

Comment:
QT ports are strange beasts. They seem to ignore the configure.compiler setting, and ignore all the complier blackisting. This port still builds with /usr/bin/clang, even though you have added this blacklisting, which is frustrating. The reason this happens is that the compiler that is used to build QT itself is hard-coded into the QT build scripting -- clang-macx I think it's called. If you want to change the compiler, you need to edit that file. 

The only solution to this is to build QT in the first place with a capable compiler, like macports-clang-5.0. Then it will use that compiler to build everything else, forever, until you replace the QT build with some other.

Reply:
Well, perhaps I should be more careful. At this moment, that's why I think the Port ignored your blacklisting. Could be there is some other reason, as I didn't test it in detail. Here's the start of the build log:
```
--->  Fetching distfiles for KeePassXC
--->  Verifying checksums for KeePassXC
--->  Checksumming keepassxc-2.3.1-src.tar.xz
--->  Extracting KeePassXC
--->  Extracting keepassxc-2.3.1-src.tar.xz
Executing:  cd "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work" && /opt/local/bin/xz -dc '/opt/local/var/macports/distfiles/KeePassXC/keepassxc-2.3.1-src.tar.xz' | /usr/bin/gnutar --no-same-owner -xf - 
--->  Applying patches to KeePassXC
--->  Applying patch-no-deployqt.diff
Executing:  cd "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1" && /usr/bin/patch -p0 < '/opt/macportsofficial/macports-ports/security/KeePassXC/files/patch-no-deployqt.diff'
patching file CMakeLists.txt
Hunk #1 succeeded at 312 (offset 6 lines).
patching file src/CMakeLists.txt
Hunk #1 succeeded at 320 (offset 1 line).
patching file src/autotype/mac/CMakeLists.txt
--->  Configuring KeePassXC
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: Qt dependency is not the latest version but may be the latest supported on your OS
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
--->  Patching CMakeLists.txt: s#/usr/local/bin#/opt/local/bin#
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Warning: All compilers are either blacklisted or unavailable; defaulting to first fallback option
Executing:  cd "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build" && /opt/local/bin/cmake -DCMAKE_INSTALL_PREFIX='/opt/local' -DCMAKE_INSTALL_PREFIX=/Applications/MacPorts -DCMAKE_INSTALL_MANDIR=/opt/local/share/man -DWITH_XC_HTTP=ON -DWITH_XC_BROWSER=ON -DWITH_XC_SSHAGENT=ON -DWITH_XC_NETWORKING=ON -DWITH_CXX11=OFF -DWITH_XC_AUTOTYPE=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_C_COMPILER="$CC" -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_INSTALL_NAME_DIR=/opt/local/lib -DCMAKE_INSTALL_RPATH=/opt/local/lib -DCMAKE_MODULE_PATH=/opt/local/share/cmake/Modules -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/opt/local;/usr" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -Wno-dev -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.7" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1 
-- The C compiler identification is AppleClang 4.2.0.4250028
-- The CXX compiler identification is AppleClang 4.2.0.4250028
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at CMakeLists.txt:53 (message):
  KeePassHTTP support has been deprecated and will be removed in a future
  version.  Please use WITH_XC_BROWSER instead!

  For enabling / disabling network access code, WITH_XC_HTTP has been
  replaced by WITH_XC_NETWORKING.


fatal: Not a git repository (or any of the parent directories): .git
-- Setting up build for KeePassXC v2.3.1

-- Performing Test WERROR_C_AVAILABLE
-- Performing Test WERROR_C_AVAILABLE - Success
-- Performing Test WERROR_CXX_AVAILABLE
-- Performing Test WERROR_CXX_AVAILABLE - Success
-- Found LibGPGError: /opt/local/lib/libgpg-error.dylib  
-- Found Gcrypt: /opt/local/lib/libgcrypt.dylib (Required is at least version "1.7.0") 
-- Found Argon2: /opt/local/lib/libargon2.dylib  
-- Found ZLIB: /opt/local/lib/libz.dylib (found version "1.2.11") 
-- Performing Test HAVE_PR_SET_DUMPABLE
-- Performing Test HAVE_PR_SET_DUMPABLE - Failed
-- Performing Test HAVE_RLIMIT_CORE
-- Performing Test HAVE_RLIMIT_CORE - Success
-- Performing Test HAVE_PT_DENY_ATTACH
-- Performing Test HAVE_PT_DENY_ATTACH - Success
-- Found CURL: /opt/local/lib/libcurl.dylib (found version "7.58.0") 
-- Found sodium: /opt/local/lib/libsodium.dylib (Required is at least version "1.0.12") 
sh: /bin/ps: Operation not permitted
-- Including translations...

-- Enabled features:
 * Networking, Compile KeePassXC with network access code (e.g. for downloading website icons)
 * KeePassXC-Browser, Browser integration with KeePassXC-Browser
 * KeePassHTTP, Browser integration compatible with ChromeIPass and PassIFox (deprecated, implies Networking)
 * SSHAgent, SSH agent integration compatible with KeeAgent

-- Disabled features:
 * Auto-Type, Automatic password typing
 * YubiKey, YubiKey HMAC-SHA1 challenge-response

-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0025
    WITH_CXX11


-- Build files have been written to: /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build
--->  Building KeePassXC
Executing:  cd "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build" && /usr/bin/make -j2 -w all VERBOSE=ON 
make: Entering directory `/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build'
/opt/local/bin/cmake -H/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1 -B/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build --check-build-system CMakeFiles/Makefile.cmake 0
/opt/local/bin/cmake -E cmake_progress_start /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/CMakeFiles /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/CMakeFiles/progress.marks
/usr/bin/make -f CMakeFiles/Makefile2 all
make[1]: Entering directory `/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build'
/usr/bin/make -f src/CMakeFiles/zxcvbn_autogen.dir/build.make src/CMakeFiles/zxcvbn_autogen.dir/depend
/usr/bin/make -f share/translations/CMakeFiles/translations.dir/build.make share/translations/CMakeFiles/translations.dir/depend
make[2]: Entering directory `/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build'
cd /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1 /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1/src /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src/CMakeFiles/zxcvbn_autogen.dir/DependInfo.cmake --color=
make[2]: Entering directory `/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build'
cd /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build && /opt/local/bin/cmake -E cmake_depends "Unix Makefiles" /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1 /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1/share/translations /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/share/translations /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/share/translations/CMakeFiles/translations.dir/DependInfo.cmake --color=
Dependee "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src/CMakeFiles/zxcvbn_autogen.dir/DependInfo.cmake" is newer than depender "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src/CMakeFiles/zxcvbn_autogen.dir/depend.internal".
Dependee "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build/src/CMakeFiles/zxcvbn_autogen.dir/depend.internal".
Scanning dependencies of target zxcvbn_autogen
```

Reply:
Yeah - what the port needed was more fallback compilers. IMHO this should not be required for a port to specify. But adding a line like this below your blacklisting line:
```
compiler.fallback-append macports-clang-5.0 macports-clang-4.0 macports-clang-3.9
```
allows the port to build with clang-5.0:
```
Executing:  cd "/opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/build" && /opt/local/bin/cmake -DCMAKE_INSTALL_PREFIX='/opt/local' -DCMAKE_INSTALL_PREFIX=/Applications/MacPorts -DCMAKE_INSTALL_MANDIR=/opt/local/share/man -DWITH_XC_HTTP=ON -DWITH_XC_BROWSER=ON -DWITH_XC_SSHAGENT=ON -DWITH_XC_NETWORKING=ON -DWITH_CXX11=OFF -DWITH_XC_AUTOTYPE=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_C_COMPILER="$CC" -DCMAKE_COLOR_MAKEFILE=ON -DCMAKE_CXX_COMPILER="$CXX" -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_INSTALL_NAME_DIR=/opt/local/lib -DCMAKE_INSTALL_RPATH=/opt/local/lib -DCMAKE_MODULE_PATH=/opt/local/share/cmake/Modules -DCMAKE_SYSTEM_PREFIX_PATH="/opt/local;/opt/local;/usr" -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -Wno-dev -DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_DEPLOYMENT_TARGET="10.7" -DCMAKE_OSX_SYSROOT="/" /opt/local/var/macports/build/_opt_macportsofficial_macports-ports_security_KeePassXC/KeePassXC/work/keepassxc-2.3.1 
-- The C compiler identification is Clang 5.0.1
-- The CXX compiler identification is Clang 5.0.1
```

This is a nutty amount of work to expect you to do, and most of this should be done behind the scenes for you in base, I believe, and it is what we are working on right now with the libc++ defaulting and appropriate compiler defaulting. 

But, to get your port working right now today, it appears that this is what is needed (in addition to the os.major test fix).